src/after-compile.ts

```diff
@@ -246,18 +246,22 @@ function provideDeclarationFilesToWebpack(
     }
 
     const outputFiles = getEmitOutput(instance, filePath);
-    const declarationFile = outputFiles
-      .filter(outputFile => outputFile.name.match(constants.dtsDtsxRegex))
-      .pop();
-    if (declarationFile !== undefined) {
-      const assetPath = path.relative(
-        compilation.compiler.context,
-        declarationFile.name
-      );
-      compilation.assets[assetPath] = {
-        source: () => declarationFile.text,
-        size: () => declarationFile.text.length
-      };
+    const declarationFiles: typescript.OutputFile[] = outputFiles.filter(
+      (outputFile: typescript.OutputFile) =>
+        outputFile.name.match(constants.dtsDtsxOrDtsDtsxMapRegex)
+    );
+
+    if (Array.isArray(declarationFiles)) {
```

Comment:
Is `Array.isArray` the right thing to do here? Presumably this will always be true as the result of the `filter` will always be an `Array`?

Reply:
Yeah, that is correct. But forEach will handle that. I'm only doing the array check to make sure forEach won't throw and error.

Reply:
Filter will always return an array. So I removed the check. It's safe to assume that we will always have an array, but it might be empty.